### PR TITLE
`case` tests

### PR DESCRIPTION
**Pull request by rktjmp**
Mostly just converts existing `match` tests to `case` and adds `(= x)` unification where needed. Adds some `case` specific tests from the documentation.

The tests are in "expanded format" for readability instead of squashed into one line (for now?)

### CODE REVIEW

File: test/macro.fnl [hunk 1]
```diff
@@ -125,6 +125,334 @@
...
+        [:a {:b b}] b)
+      8)
+  (== (case (values 1 2 3 4 :ok)
+        (where (a b c d e) (= 1 a)) e
+        _ :not-ok) "ok")
+  (== (case [:a :b :c]
+        [a b c] (.. b :eee))
+      "beee")
+
+  ;; TODO: should fail to compile?
```
Comment (XeroOl) [inline test/macro.fnl:149]: I think this should fail to compile

File: test/macro.fnl [hunk 2]
```diff
@@ -499,6 +499,31 @@
...
+        (where [a b] (<= a 20)) true
+        (catch
+          a :ay ;; should not get here
+          _ :nothing))
+     true)
+
+  ;; ensure we do not unify in a catch
+  ;; these can be sense checked by running match-try with fresh bindings at
+  ;; each step
+  (== (catch-try 20
```
Comment (XeroOl) [inline test/macro.fnl:518]: this should say `case-try`

File: test/macro.fnl (hunk 1 above)
Comment (rktjmp) [reply test/macro.fnl]: Added test

File: test/macro.fnl [hunk 3]
```diff
@@ -125,6 +125,400 @@
...
+          ;; [99 20] = [99 b]
+          [(= x) b] (+ x b)))
+     (+ 99 20))
+
+  (== (let [x 99]
+        (case [99 20]
+          ;; [99 20] = [99 x]
+          ;; note that x is bound in the pattern, so the body uses that value
+          [(= x) x] (+ x x)))
+     (+ 20 20)))
```
Comment (technomancy) [inline test/macro.fnl:495]: Missing a close paren here.

Looks like this could be rebased on main now?

**Comment by rktjmp** [comment]
I rebased it on https://git.sr.ht/~technomancy/fennel, and seems it has caught some not-yet-githubed commits.

Should I rebase it on gh:main instead or ?

**Comment by technomancy** [comment]
Oops; looks like the two remotes got out of sync. I've brought them back to parity, so you should be god now.

**Comment by technomancy** [comment]
Looks like the tests are still failing here.

**Comment by rktjmp** [comment]
Sorry! Wasn't actually .. testing.. the tests as I wrote them blind before the feature - miscommunication, thought they'd been proofed with the feature patch. They should all be passing now.

**Comment by technomancy** [comment]
Haha no worries; thanks!

**Pull request merged**